botocore/endpoint.py

```diff
@@ -227,6 +229,18 @@ def _get_response(self, request, operation_model, attempts):
                 request, verify=self.verify,
                 stream=operation_model.has_streaming_output,
                 proxies=self.proxies, timeout=self.timeout)
+        except ConnectionError as e:
+            # For a connection error, if it looks like it's a DNS
+            # lookup issue, 99% of the time this is due to a misconfigured
+            # region/endpoint so we'll raise a more specific error message
+            # to help users.
+            if self._looks_like_dns_error(e):
+                endpoint = e.request.url
```

Comment:
Ok I'll get this updated.  I think I may send a PR to the CLI that catches this error and gives a better error message though.

The high level thing I'm after is that in the CLI, if a user types a wrong region name, e.g. `--region eu-central-typo`, we can give then some type of hint that tells them to verify their region name.  I agree that this part in botocore isn't the right place to have this error message.